File: KataFour/KataFourTest.py
import unittest
import KataFour


class Test(unittest.TestCase):
    def test_weather(self):
        self.assertEqual('14', KataFour.smallestTempSpread())

    def test_football(self):
        self.assertEqual('Aston_Villa', KataFour.smallestScoreSpread())

    def test_weather_refactor(self):
        self.assertEqual('14', KataFour.smallestTempRefactor())

    def test_football_refactor(self):
        self.assertEqual('Aston_Villa', KataFour.smallestScoreRefactor())


if __name__ == "__main__":
    unittest.main()

File: ruby/04/KataFourTest.rb
require 'test/unit'
require 'KataFour'


class KataFourTest < Test::Unit::TestCase
    def test_weather
        assert_equal('14', KataFour.smallest_temp_spread())
    end
    
    def test_football
        assert_equal('Aston_Villa', KataFour.smallest_score_spread())
    end
    
    def test_weather_refactor
        assert_equal('14', KataFour.smallest_temp_refactor())
    end
    
    def test_football_refactor
        assert_equal('Aston_Villa', KataFour.smallest_score_refactor())
    end                                                                           
end
